pkg/controller/storagecluster/initialization_reconciler.go

```diff
@@ -297,6 +298,77 @@ func (r *ReconcileStorageCluster) newStorageClasses(initData *ocsv1.StorageClust
 	return ret, nil
 }
 
+// newRBDSnapshotClass returns a new RBD backed SnapshotClass
+func (r *ReconcileStorageCluster) newRBDSnapshotClass(initData *ocsv1.StorageCluster) *snapapi.VolumeSnapshotClass {
+	retSC := &snapapi.VolumeSnapshotClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: generateNameForRBDSnapshotClass(initData),
+		},
+		Driver: generateNameForRBDSnapshotClassDriver(initData),
+		Parameters: map[string]string{
+			"clusterID": initData.Namespace,
+			"csi.storage.k8s.io/snapshotter-secret-name":      "rook-csi-rbd-provisioner",
+			"csi.storage.k8s.io/snapshotter-secret-namespace": initData.Namespace,
+		},
+		DeletionPolicy: snapapi.VolumeSnapshotContentDelete,
+	}
+	return retSC
+}
+
+func (r *ReconcileStorageCluster) newSnapshotClasses(instance *ocsv1.StorageCluster) []*snapapi.VolumeSnapshotClass {
+	scs := []*snapapi.VolumeSnapshotClass{
+		r.newRBDSnapshotClass(instance),
+		// TODO
+		// r.newCephFsSnapshotClass(instance),
+	}
+	return scs
+}
+
+func (r *ReconcileStorageCluster) createSnapshotClasses(scs []*snapapi.VolumeSnapshotClass, reqLogger logr.Logger) error {
+	for _, sc := range scs {
+		existing := &snapapi.VolumeSnapshotClass{}
+		if err := r.client.Get(context.TODO(), types.NamespacedName{Name: sc.Name, Namespace: sc.Namespace}, existing); err != nil {
```

Comment:
I think this can be rewritten to this format. 

directly create() -> get err -> validate err.AlreadyExist() or something else. That way the error handling would be pretty clean or neat.

Reply:
I have changed the error handling slightly in the new code.

IMHO, the process you described (above), is a complete negation of what we currently do ,
`Get()` -> err check -> if `err == NotFound()` -> `Create()` -> throw any creation error
and it is just a perspective difference (like glass half empty or half full). (=

Please let me know, if I understood your view-point completely wrong.

Reply:
thats interesting.. iic, the general practice what I have seen is, create() and check for errors, that would reduce one call to api server.. Not sure why the other pattern is followed here . @jarrpa @obnoxxx @umangachapagain @travisn thoughts ?

Reply:
Current format is what we typically do in the ocs-operator code, so LGTM.